src/OpenTelemetry.Extensions.EventSource/OpenTelemetryEventSourceLogEmitter.cs

```diff
@@ -0,0 +1,224 @@
+            this.openTelemetryLoggerProvider = openTelemetryLoggerProvider!;
+            this.disposeProvider = disposeProvider;
+            this.shouldListenToFunc = shouldListenToFunc;
+
+            var logEmitter = this.openTelemetryLoggerProvider.CreateEmitter();
+            Debug.Assert(logEmitter != null, "logEmitter was null");
+
+            this.logEmitter = logEmitter!;
+
+            lock (this.lockObj)
+            {
+                foreach (EventSource eventSource in this.eventSourcesBeforeConstructor)
+                {
+                    this.ProcessSource(eventSource);
+                }
+
+                this.eventSourcesBeforeConstructor = null;
+            }
+        }
+
+        /// <inheritdoc/>
+        public override void Dispose()
+        {
+            foreach (EventSource eventSource in this.eventSources)
+            {
+                this.DisableEvents(eventSource);
+            }
+
+            this.eventSources.Clear();
+
+            if (this.disposeProvider)
+            {
+                this.openTelemetryLoggerProvider.Dispose();
+            }
+
+            base.Dispose();
+        }
+
+#pragma warning disable CA1062 // Validate arguments of public methods
+        /// <inheritdoc/>
+        protected override void OnEventSourceCreated(EventSource eventSource)
+        {
+            Debug.Assert(eventSource != null, "EventSource was null.");
+
+            try
+            {
+                if (this.eventSourcesBeforeConstructor != null)
+                {
+                    lock (this.lockObj)
+                    {
+                        if (this.eventSourcesBeforeConstructor != null)
+                        {
+                            this.eventSourcesBeforeConstructor.Add(eventSource!);
+                            return;
+                        }
+                    }
+                }
+
+                this.ProcessSource(eventSource!);
+            }
+            finally
+            {
+                base.OnEventSourceCreated(eventSource);
+            }
+        }
+#pragma warning restore CA1062 // Validate arguments of public methods
+
+#pragma warning disable CA1062 // Validate arguments of public methods
+        /// <inheritdoc/>
+        protected override void OnEventWritten(EventWrittenEventArgs eventData)
+        {
+            Debug.Assert(eventData != null, "EventData was null.");
+
+            string? rawMessage = eventData!.Message;
+
+            LogRecordData data = new(Activity.Current)
+            {
+#if NETSTANDARD2_1_OR_GREATER
+                Timestamp = eventData.TimeStamp,
+#endif
+                EventId = new EventId(eventData.EventId, eventData.EventName),
+                LogLevel = ConvertEventLevelToLogLevel(eventData.Level),
+            };
+
+            LogRecordAttributeList attributes = default;
+
+            attributes.Add("event_source.name", eventData.EventSource.Name);
```

Comment:
@cijothomas @alanwest @reyang @utpilla Right now EventSource.Name becomes the event_source.name tag. CategoryName is null. Anyone feel like EventSource.Name should end up the CategoryName and we should drop the tag? I kind of feel like that would be a good change.

Reply:
hmm. Well the goofy thing about `CategoryName` is that it is an `ILogger` concept. We've taken the stance to kind of treat ILogger as special in OpenTelemetry libraries. I'm not sure it makes sense to try to coerce other logging libraries concepts into `ILogger` concepts.

From the standpoint of OTLP we currently transform `CategoryName` into an attribute named `dotnet.ilogger.category`.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/cf3dfc51f589dc1a172d056cfab604b0710c14f8/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs#L78-L86

It might actually make more sense to abandon the use of `CategoryName` and add an appropriately named attribute to the LogRecord upstream just like you've done here with `event_source.name`

Reply:
Also regarding the comment 

> // 1.Track the following issue, and map CategoryName to Name 
     // if it makes it to log data model.

The `Name` field _did not_ make it into the data model: https://github.com/open-telemetry/opentelemetry-proto/pull/357

Reply:
You convinced me to just leave it alone (as event_source.name tag).

FYI in the Serilog sink I do have this code...

https://github.com/open-telemetry/opentelemetry-dotnet/blob/bfabe9bc26f904860c1e09541f5abbf6febe8bf2/src/OpenTelemetry.Extensions.Serilog/OpenTelemetrySerilogSink.cs#L71-L75

...which will set `CategoryName` when users use the `ForContext` API...

```csharp
ILogger programLogger = Log.Logger.ForContext<Program>(); // <- CategoryName will be essentially typeof(Program).FullName
```

Seemed like a good fit to me but we may want to change that?

Reply:
Sorry, didn't see your comment here... yes, we need to consider what to do. Opened #3491 for discussion.

I'm not familiar enough with Serilog to make a suggestion here. Seems like a `serilog.context` attribute may make sense? Though, a quick search over Serilog code leads me to believe `ForContext` may be a bit more complex than just a name - look at this for example: https://github.com/serilog/serilog/blob/697a23189e3068bfe0da1460a2676913239d2757/src/Serilog/LoggerExtensions.cs#L24-L29... though maybe this is more akin to ILogger scopes?